<!-- Launch file for the rrt-detector and the assigner -->


<launch>
<arg name="eta" value="2.0"/>
<arg name="Geta" value="15.0"/>
<arg name="map_topic" value="/local_map/local_map"/>


  <node pkg="rrt_exploration" type="global_rrt_detector" name="global_detector" output="screen">
  <param name="eta" value="$(arg Geta)"/>
  <param name="map_topic" value="$(arg map_topic)"/>
  </node>
  
  <node pkg="rrt_exploration" type="local_rrt_detector" name="local_detector" output="screen">
  <param name="eta" value="$(arg eta)"/>
  <param name="map_topic" value="$(arg map_topic)"/>
  <param name="robot_frame" value="/odom"/>
  </node>

  <node pkg="rrt_exploration" type="filter" name="merge_filter" output="screen">
  <param name="map_topic" value="$(arg map_topic)"/>
  <param name="min_frontier_size" value="1"/>
  <param name="min_frontier_dist" value="2"/>
  <param name="goals_topic" value="/detected_points"/>
  <param name="rate" value="100"/>
  </node>
  
  <node pkg="rrt_exploration" type="another_assigner.py" name="assigner" output="screen">
  <param name="map_topic" value="$(arg map_topic)"/>
  <param name="info_radius" value="1"/> 
  <param name="info_multiplier" value="3.0"/> 
  <param name="hysteresis_radius" value="10.0"/>
  <param name="hysteresis_gain" value="2.0"/> 
  <param name="frontiers_topic" value="/filtered_points"/> 
  <param name="n_robots" value="1"/>
  <param name="delay_after_assignement" value="0.5"/>
  <param name="rate" value="100"/>
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rrt_exploration)/rviz/rrt_cpp_detector.rviz"/>


  
</launch>
